From: Reiner Herrmann <reiner@reiner-h.de>
Date: Sat, 21 Dec 2019 22:31:04 +0100
Subject: Improve autoreconf/autoheader support

Forwarded: https://github.com/michaelrsweet/htmldoc/pull/329

autoheader complained about a bunch "missing templates", because AC_DEFINE used
an older syntax, which failed the autoreconf step.
---
 configure.ac | 52 +++++++++++++++++++++++++++-------------------------
 1 file changed, 27 insertions(+), 25 deletions(-)

diff --git a/configure.ac b/configure.ac
index e9e285b..1faab58 100644
--- a/configure.ac
+++ b/configure.ac
@@ -28,7 +28,7 @@ AC_CONFIG_HEADERS([config.h])
 dnl De])ne the version number...
 SVERSION="AC_PACKAGE_VERSION"
 AC_SUBST(SVERSION)
-AC_DEFINE_UNQUOTED(SVERSION, "$SVERSION")
+AC_DEFINE_UNQUOTED([SVERSION], ["$SVERSION"], [version number])
 
 
 dnl GNU autoheader, automake, etc. will break the build system, this macro
@@ -103,8 +103,8 @@ AS_IF([test "$ac_cv_prog_ranlib" = ":"], [
 AC_SUBST(ARFLAGS)
 
 dnl Checks for header files.
-AC_CHECK_HEADER(strings.h, AC_DEFINE(HAVE_STRINGS_H))
-AC_CHECK_HEADER(locale.h, AC_DEFINE(HAVE_LOCALE_H))
+AC_CHECK_HEADER(strings.h, AC_DEFINE([HAVE_STRINGS_H], [], [strings.h support]))
+AC_CHECK_HEADER(locale.h, AC_DEFINE([HAVE_LOCALE_H], [], [locale.h support]))
 
 dnl Checks for string functions.
 AC_CHECK_FUNCS(strdup strcasecmp strncasecmp strlcat strlcpy snprintf vsnprintf)
@@ -119,7 +119,7 @@ AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#include <time.h>]], [[
     int o = t.tm_gmtoff;
 ]])], [
     AC_MSG_RESULT([yes])
-    AC_DEFINE(HAVE_TM_GMTOFF)
+    AC_DEFINE([HAVE_TM_GMTOFF], [], [tm_gmtoff support])
 ], [
     AC_MSG_RESULT([no])
 ])
@@ -129,17 +129,16 @@ LDFLAGS="${LDFLAGS:=}"
 AC_SUBST(LDFLAGS)
 
 AC_CHECK_LIB(m,pow)
-AC_CHECK_FUNC(poll, AC_DEFINE(HAVE_POLL))
+AC_CHECK_FUNC(poll, AC_DEFINE([HAVE_POLL], [], [poll support]))
 AC_SEARCH_LIBS(socket, socket)
 AC_SEARCH_LIBS(gethostbyaddr, nsl)
-AC_SEARCH_LIBS(getaddrinfo, nsl, AC_DEFINE(HAVE_GETADDRINFO))
-AC_SEARCH_LIBS(getnameinfo, nsl, AC_DEFINE(HAVE_GETNAMEINFO))
-AC_SEARCH_LIBS(hstrerror, nsl socket resolv, AC_DEFINE(HAVE_HSTRERROR))
-AC_SEARCH_LIBS(__res_init, resolv bind, AC_DEFINE(HAVE_RES_INIT), [
+AC_SEARCH_LIBS(getaddrinfo, nsl, AC_DEFINE([HAVE_GETADDRINFO], [], [getaddrinfo support]))
+AC_SEARCH_LIBS(getnameinfo, nsl, AC_DEFINE([HAVE_GETNAMEINFO], [], [getnameinfo support]))
+AC_SEARCH_LIBS(hstrerror, nsl socket resolv, AC_DEFINE([HAVE_HSTRERROR], [], [hstrerror support]))
+AC_SEARCH_LIBS(__res_init, resolv bind, AC_DEFINE([HAVE_RES_INIT], [], [__res_init support]),
     AC_SEARCH_LIBS(res_9_init, resolv bind, AC_DEFINE(HAVE_RES_INIT),
-    AC_SEARCH_LIBS(res_init, resolv bind, AC_DEFINE(HAVE_RES_INIT)))
-])
-AC_CHECK_HEADER(resolv.h, AC_DEFINE(HAVE_RESOLV_H))
+    AC_SEARCH_LIBS(res_init, resolv bind, AC_DEFINE(HAVE_RES_INIT))))
+AC_CHECK_HEADER(resolv.h, AC_DEFINE([HAVE_RESOLV_H], [], [resolv.h support]))
 
 dnl Check for largefile support...
 AC_SYS_LARGEFILE
@@ -175,10 +174,10 @@ AC_CACHE_CHECK([for long long int], ac_cv_c_long_long, [
 ])
 
 AS_IF([test $ac_cv_c_long_long = yes], [
-    AC_DEFINE(HAVE_LONG_LONG)
+    AC_DEFINE([HAVE_LONG_LONG], [], [long long support])
 ])
 
-AC_CHECK_FUNC(strtoll, AC_DEFINE(HAVE_STRTOLL))
+AC_CHECK_FUNC(strtoll, AC_DEFINE([HAVE_STRTOLL], [], [strtoll support]))
 
 dnl Check for TLS/SSL libraries...
 AC_ARG_ENABLE(ssl, AS_HELP_STRING([--disable-ssl], [turn off SSL/TLS support, default=enabled]))
@@ -195,13 +194,16 @@ AS_IF([test x$enable_ssl], [
 	AS_IF([test $host_os_name = darwin], [
 	    AC_CHECK_HEADER(Security/SecureTransport.h, [
 	    	have_ssl=1
-		AC_DEFINE(HAVE_SSL)
-		AC_DEFINE(HAVE_CDSASSL)
+		AC_DEFINE([HAVE_SSL], [], [ssl support])
+		AC_DEFINE([HAVE_CDSASSL], [], [cdsassl support])
 		SSLLIBS="-framework Security -framework CoreFoundation"
 
-		AC_CHECK_HEADER(Security/SecCertificate.h, AC_DEFINE(HAVE_SECCERTIFICATE_H))
-		AC_CHECK_HEADER(Security/SecItem.h, AC_DEFINE(HAVE_SECITEM_H))
-		AC_CHECK_HEADER(Security/SecPolicy.h, AC_DEFINE(HAVE_SECPOLICY_H))
+		AC_CHECK_HEADER(Security/SecCertificate.h,
+		    AC_DEFINE([HAVE_SECCERTIFICATE_H], [], [SecCertificate.h support]))
+		AC_CHECK_HEADER(Security/SecItem.h,
+		    AC_DEFINE([HAVE_SECITEM_H], [], [SecItem.h support]))
+		AC_CHECK_HEADER(Security/SecPolicy.h,
+		    AC_DEFINE([HAVE_SECPOLICY_H], [], [SecPolicy.h support]))
 	    ])
 	])
     ])
@@ -215,20 +217,20 @@ AS_IF([test x$enable_ssl], [
 	    SSLLIBS=`$PKGCONFIG --libs gnutls`
 	    SSLFLAGS=`$PKGCONFIG --cflags gnutls`
 	    AC_DEFINE(HAVE_SSL)
-	    AC_DEFINE(HAVE_GNUTLS)
+	    AC_DEFINE([HAVE_GNUTLS], [], [gnutls support])
 	], [test "x$LIBGNUTLSCONFIG" != x], [
 	    have_ssl=1
 	    SSLLIBS=`$LIBGNUTLSCONFIG --libs`
 	    SSLFLAGS=`$LIBGNUTLSCONFIG --cflags`
 	    AC_DEFINE(HAVE_SSL)
-	    AC_DEFINE(HAVE_GNUTLS)
+	    AC_DEFINE([HAVE_GNUTLS], [], [gnutls support])
 	])
 
 	AS_IF([test $have_ssl = 1], [
 	    SAVELIBS="$LIBS"
 	    LIBS="$LIBS $SSLLIBS"
-	    AC_CHECK_FUNC(gnutls_transport_set_pull_timeout_function, AC_DEFINE(HAVE_GNUTLS_TRANSPORT_SET_PULL_TIMEOUT_FUNCTION))
-	    AC_CHECK_FUNC(gnutls_priority_set_direct, AC_DEFINE(HAVE_GNUTLS_PRIORITY_SET_DIRECT))
+	    AC_CHECK_FUNC(gnutls_transport_set_pull_timeout_function, AC_DEFINE([HAVE_GNUTLS_TRANSPORT_SET_PULL_TIMEOUT_FUNCTION], [], [gnutls transport_set_pull_timeout_function support]))
+	    AC_CHECK_FUNC(gnutls_priority_set_direct, AC_DEFINE([HAVE_GNUTLS_PRIORITY_SET_DIRECT], [], [gnutls priority_set_direct support]))
 	    LIBS="$SAVELIBS"
 	])
     ])
@@ -325,8 +327,8 @@ AS_IF([test "$datadir" = "\${prefix}/share"], [
     datadir="$datarootdir"
 ])
 
-AC_DEFINE_UNQUOTED(DOCUMENTATION, "$datadir/doc/htmldoc")
-AC_DEFINE_UNQUOTED(HTML_DATA, "$datadir/htmldoc")
+AC_DEFINE_UNQUOTED([DOCUMENTATION], ["$datadir/doc/htmldoc"], [documentation directory])
+AC_DEFINE_UNQUOTED([HTML_DATA], ["$datadir/htmldoc"], [data directory])
 
 dnl Update compiler options...
 AC_ARG_ENABLE(maintainer, AS_HELP_STRING([--enable-maintainer], [turn on maintainer mode, default=no]))
